static/js/stream_edit.js

```diff
@@ -686,7 +684,56 @@ exports.initialize = function () {
         e.stopPropagation();
     });
 
-    $("#subscriptions_table").on("click", "#change-stream-privacy-button", change_stream_privacy);
+    $("#subscriptions_table").on(
+        "click",
+        ".subsection-header .subsection-changes-save .button",
+        change_stream_permissions,
+    );
+
+    $("#subscriptions_table").on(
+        "click",
+        ".subsection-header .subsection-changes-discard .button",
+        (e) => {
+            e.preventDefault();
+            e.stopPropagation();
+            const save_btn_controls = $(e.target).closest(".save-button-controls");
+            settings_org.change_save_button_state(save_btn_controls, "discarded");
+        },
+    );
+
+    $("#subscriptions_table").on("change", "#stream_privacy_modal input[name='privacy']", (e) => {
+        e.stopPropagation();
+        e.preventDefault();
+        const subsection = $(e.target).closest(".org-subsection-parent");
+        exports.save_discard_widget_status_handler(subsection);
+    });
+
+    $("#subscriptions_table").on(
+        "change",
+        "#stream_privacy_modal input[name='stream-post-policy']",
+        (e) => {
+            e.stopPropagation();
+            e.preventDefault();
+            const subsection = $(e.target).closest(".org-subsection-parent");
+            exports.save_discard_widget_status_handler(subsection);
+        },
+    );
+
+    $("#subscriptions_table").on("change", "#stream_message_retention", (e) => {
+        e.stopPropagation();
+        e.preventDefault();
+        const message_retention_setting_dropdown_value = e.target.value;
+        if (
+            message_retention_setting_dropdown_value === "retain_for_period" &&
+            $(e.target).attr("name") === "stream_message_retention_setting"
+        ) {
+            // Dont submit on this change as number of days have not been entered yet.
+            return;
+        }
+
+        const subsection = $(e.target).closest(".org-subsection-parent");
+        exports.save_discard_widget_status_handler(subsection);
+    });
```

Comment:
I think we can improve here, here we have a lot of code duplication, it will be a good case to use/create a class for the selector for these three event handlers. Have you tried and faced any problem with this, if not, then please try and let us know?